eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojoWithObjectionaries.java

```diff
@@ -0,0 +1,111 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.cactoos.scalar.Unchecked;
+import org.eolang.maven.hash.CommitHash;
+import org.eolang.maven.objectionary.Objectionary;
+
+/**
+ * {@link SafeMojo} with objectionaries.
+ *
+ * @since 0.29.6
+ */
+abstract class SafeMojoWithObjectionaries extends SafeMojo {
+    /**
+     * The Git hash to pull objects from, in objectionary.
+     *
+     * @since 0.21.0
+     * @checkstyle VisibilityModifierCheck (6 lines)
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(property = "eo.tag", required = true, defaultValue = "master")
+    protected String tag = "master";
+
+    /**
+     * Read hashes from local file.
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     * @checkstyle VisibilityModifierCheck (6 lines)
+     */
+    @Parameter(property = "offlineHashFile")
+    protected Path offlineHashFile;
+
+    /**
+     * Return hash by pattern.
+     * -DofflineHash=0.*.*:abc2sd3
+     * -DofflineHash=0.2.7:abc2sd3,0.2.8:s4se2fe
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     * @checkstyle VisibilityModifierCheck (6 lines)
+     */
+    @Parameter(property = "offlineHash")
+    protected String offlineHash;
+
+    /**
+     * The objectionary.
+     *
+     * @checkstyle VisibilityModifierCheck (5 lines)
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    protected Objectionary objectionary;
+
+    /**
+     * Hash-Objectionary map.
+     * @todo #1602:30min Use objectionaries to pull objects with different
+     *  versions. Objects with different versions are stored in different
+     *  storages (objectionaries). Every objectionary hash its own hash.
+     *  To pull versioned object from objectionary firstly we need to get
+     *  right objectionary by object's version and then get object from that
+     *  objectionary by name.
+     * @todo #1602:30min Use objectionaries to probe objects with different
+     *  versions. Objects with different versions are stored in different
+     *  storages (objectionaries). Every objectionary has its own hash.
+     *  To get versioned object from objectionary firstly we need to get
+     *  right objectionary by object's version and then get object from that
+     *  objectionary by name.
+     * @checkstyle MemberNameCheck (5 lines)
+     * @checkstyle VisibilityModifierCheck (4 lines)
+     */
+    protected final Map<String, Objectionary> objectionaries = new HashMap<>();
```

Comment:
@maxonfjvipon Looks like we can have only one `hash` (a `CommitHash` instance), so I don't clearly understand why we need that map to save the only one entry?

Reply:
@volodya-lombrozo This PR is just preparation. Isn't it clear from `todo` what this map was added for?